### PR TITLE
[Enhancement](column) support default value for int/bigint/decimal type column

### LINKED ISSUE #20670: [Enhancement] default value for bigint should not has to be quoted
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

CREATE TABLE t (topic BIGINT DEFAULT 0) does not work. CREATE TABLE t (topic BIGINT DEFAULT "0") works. 

### Solution

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sjyango**
## Proposed changes

Issue Number: close https://github.com/apache/doris/issues/20670 https://github.com/apache/doris/issues/24125

Support int/bigint/decimal default value to int/bigint/decimal type column.
legacy work: https://github.com/apache/doris/pull/23594

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sjyango** [comment]
run buildall

**Comment by sjyango** [comment]
run feut

File: fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4 [hunk 1]
```diff
@@ -579,8 +579,8 @@ columnDef
         (aggType=aggTypeDef)?
         ((NOT)? NULL)?
         (AUTO_INCREMENT (LEFT_PAREN autoIncInitValue=number RIGHT_PAREN)?)?
-        (DEFAULT (nullValue=NULL | INTEGER_VALUE | stringValue=STRING_LITERAL| CURRENT_DATE
-            | defaultTimestamp=CURRENT_TIMESTAMP (LEFT_PAREN defaultValuePrecision=number RIGHT_PAREN)?))?
+        (DEFAULT (nullValue=NULL | INTEGER_VALUE | DECIMAL_VALUE | stringValue=STRING_LITERAL |
+         CURRENT_DATE | defaultTimestamp=CURRENT_TIMESTAMP (LEFT_PAREN defaultValuePrecision=number RIGHT_PAREN)?))?
```
Comment (morrySnow) [inline fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4:583]:
```suggestion
        (DEFAULT (nullValue=NULL | INTEGER_VALUE | DECIMAL_VALUE | stringValue=STRING_LITERAL
           | CURRENT_DATE | defaultTimestamp=CURRENT_TIMESTAMP (LEFT_PAREN defaultValuePrecision=number RIGHT_PAREN)?))?
```

**Comment by morrySnow** [comment]
If this request is not urgent, please wait for the 'generated' column feature to solve all problems at once

**Comment by sjyango** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**